Update `@typescipt-eslint/*` packages (#433)
Shopify/eslint-plugin-shopify: README.md

}
```

If you are working on an ES5 project, extend the ES5 version of the configuration:

```json
{
  "extends": "plugin:shopify/es5"
}
```

You can also add some "augmenting" configs on top of the "core" config by extending an array of linting configs. For example, the following configuration would provide a base ESNext config that is augmented by a jest config:

```json
{
  "extends": [
    "plugin:shopify/esnext",

>     "plugin:shopify/jest"

Reviewer: I'd rather we use another config that should apply universally (IMO most projects should have the jest rule isolated to an `override`)

  ]
}
```

> Likewise, if you are using TypeScript and React, the following configuration extends the TypeScript base config with the React-specific rules provided by the React configuration file:

Author: ```suggestion
Likewise, if you are using TypeScript and React, the following example extends the TypeScript base configuration with the React-specific rules provided by the React configuration:
```

This plugin provides the following core configurations:

- [esnext](lib/config/esnext.js): Use this for anything written with ES2015+ features.

> - [typescript](lib/config/typescript.js): Use this for Typescript projects. The rules enabled in this confige do not require type-checking to run. To enable all Typescript rules, you must augment this config with the `typescript-requiring-type-checking` config mentioned below.

Author: ```suggestion
- [typescript](lib/config/typescript.js): Use this for Typescript projects. This configuration enables a subset of Typescript-specific that do not require type-checking to run. To enable all Typescript rules, you must augment this config with the `typescript-requiring-type-checking` config mentioned below.
```

- [es5](lib/config/es5.js): Use this for legacy projects.

This plugin also provides the following tool-specific configurations, which can be used on top of the core configurations:

> - [typescript-requiring-type-checking](lib/config/typescript-requiring-type-checking.js) Use this config to augment the `typescript` config to enable all TypeScript rules, including those that require type checking. These rules are slower to run and and you will need to specify a path to your tsconfig.json file in the "project" property of "parserOptions". The following example would provide all of the TypeScript rules, assuming the tsconfig.json is in the same directory as you ESlint configuration.

Author: ```suggestion
- [typescript-requiring-type-checking](lib/config/typescript-requiring-type-checking.js) Use this to augment the `typescript` config and enable all TypeScript rules, including those that require type checking. These rules are slower to run, and you will need to specify a path to your tsconfig.json file in the "project" property of "parserOptions". The following example would provide all of the TypeScript rules, assuming the tsconfig.json is in the same directory as your ESlint configuration.
```

> - [typescript-requiring-type-checking](lib/config/typescript-requiring-type-checking.js) Use this to augment the `typescript` config and enable all TypeScript rules, including those that require type checking. These rules are slower to run, and you will need to specify a path to your tsconfig.json file in the "project" property of "parserOptions". The following example would provide all of the TypeScript rules, assuming the tsconfig.json is in the same directory as your ESlint configuration.

Reviewer: I would drop `-requiring` from the name

Reviewer 2: Alternative naming bikeshedding: `typescript-extended` or `typescript-advanced`?
